src/components/LocationPage/ShareButtons.tsx

```diff
@@ -26,30 +27,28 @@ const ShareButtons: React.FC<{
     urlJoin(shareBaseURL, `chart/${chartIdentifier}`),
   );
 
-  const [showShareIcons, setShowShareIcons] = useState(false);
+  const [showSocialButtons, setShowSocialButtons] = useState(false);
 
-  // Delay allows the user to briefly see copy-link button text change when clicked
   const hideSocialButtons = () => {
-    const timeoutId = setTimeout(() => {
-      setShowShareIcons(false);
-    }, 1500);
+    const timeoutId = setTimeout(() => setShowSocialButtons(false), 1500);
```

Comment:
It looks like the `ShareButtons` component doesn't need the delay at all, maybe we should pass `setShowSocialButtons` to the `SocialButtons` as prop instead?

```tsx
<SocialButtons
  {/* ... */}
  region={region}
  setShowSocialButtons={setShowSocialButtons}
/>
```

and then in `SocialButtons` we can use [`useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect) to update the state with a delay. 

> Mutations, subscriptions, timers, logging, and other side effects are not allowed inside the main body of a function component (referred to as React’s render phase). Doing so will lead to confusing bugs and inconsistencies in the UI.

```tsx
// timers are not allowed in the main body of react components, they must be
// wrapped with useEffect
useEffect(() => {
  const timeoutId = setTimeout(() => setShowSocialButtons(false), 500);
  return () => clearTimeout(timeoutId);
});

// we probably don't need the delay when the user closes with esc
useEscToClose(() => setShowSocialButtons(false));
```

Alternatively, we could just pass a `closeMenu` function since we are only closing the menu from `SocialButtons`.


```tsx
<SocialButtons
  {/* ... */}
  region={region}
  closeMenu={() => setShowSocialButtons(false)}
/>
```

and then

```tsx
useEffect(() => {
  const timeoutId = setTimeout(closeMenu, 500);
  return () => clearTimeout(timeoutId);
});

// we probably don't need the delay when the user closes with esc
useEscToClose(closeMenu);
```